### PR TITLE
[mds-metrics] Fix binning bug

### PR DESCRIPTION
**Pull request by avatarneil**
This is what happens when you don't have tests yet ;)

## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [x] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author
- [x] Metrics

### CODE REVIEW

File: packages/mds-metrics/utils.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import { now, yesterday } from '@mds-core/mds-utils'
+
 import { GetTimeBinsParams } from './types'
 
-export function getTimeBins({
+export async function getTimeBins({
```
Comment (avatarneil) [inline packages/mds-metrics/utils.ts]: Weirdly enough, I discovered not marking this function as `async` and `awaiting` it causes a whole bundle of issues... I'm guessing some weirdness with the array constructor method.

File: packages/mds-metrics/utils.ts [hunk 2]
```diff
@@ -8,9 +9,7 @@ export function getTimeBins({
 }: Partial<GetTimeBinsParams>) {
   const interval = end_time - start_time
 
-  const bins = new Array(Math.floor(interval / bin_size))
-
-  return bins.map((_, idx) => ({
+  return [...Array(Math.floor(interval / bin_size))].map((_, idx) => ({
```
Comment (macsj200) [inline packages/mds-metrics/utils.ts]: Do we need to `...` spread this, or can we just call `Array().map()`?

Comment (avatarneil) [reply packages/mds-metrics/utils.ts]: Need to spread it. If you don't, it creates an array of the specified length, but doesn't actually have any items. So, when you try to call `.map()` on it, nothin' happens!

Comment (drtyh2o) [reply packages/mds-metrics/utils.ts]: Another alternative is to use `Array.from` which supports a map function over an _array-like_ object, i.e.
```
Array.from({ length: Math.floor(interval / bin_size) }, (_,idx) => { ... })
```

Comment (macsj200) [reply packages/mds-metrics/utils.ts]: I have no strong opinion either way, I just wanted to understand what it was doing

Comment (drtyh2o) [reply packages/mds-metrics/utils.ts]: Me either, but it seems `Array.from` was designed for this sort of use case.

**Pull request merged**